src/App.tsx

```diff
@@ -1,14 +1,24 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { useTodoContext } from './components/context/TodoContext';
+import { useTodoModal } from './hooks/useTodoModal';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const { originalTodos } = useTodoContext();
+  const { todoModal, userModal, todoModalId, setTodoModalId, closeModal } =
+    useTodoModal();
```

Comment:
Tutaj taki protip na przyszłosc. Pomysl o sytuacji w ktorej twoja aplikacja sie rozrasta i bedziesz chciał użyć modala na przykład nie wiem przy rejestracji użytkownika. Dziwnie bedzie uzywac w tamtym miejscu komponentu TodoModal bo wiążę się on strikte z funkcjonalnoscią jaką jest Todo a nie Rejestracja użytkownika. 

Reasumujac dobrze myslec o takich komponentach jako o odzielnych klocuszkach które maja myslec same o sobie. Oczywiscie w tej aplikacji jest to jak najbardziej ok bo jest ona bardzo mała i wiesz ze nie bedzie sie rozwijać.